src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala

```diff
@@ -13,11 +14,14 @@ trait WorkerMain[S] {
 
   protected[this] def init(args: Option[Array[String]]): S
```

Comment:
I think this is still useful for when you need to initialize the worker.

`init` happens once when the worker starts. `work` happens on every work request.

Agree that it is called in frequently and a good tradeoff may be providing a default implementation of init, but I'm not sure off the top of my head how to accomplish that because of the generics.

Reply:
I was mainly concerned with us re-parsing arguments in `work` in a lot of places. I wonder if removing the `args` parameter in `work` would help to enforce that we're doing argument parsing in `init`.

Reply:
Argument parsing happens in work, though. In `work`, we parse the arguments for each work request.

In `init` we parse args for the worker as a whole. I think the only place we have a valid use case for this is the Zinc incremental compilation stuff, which I'd like to remove. Beyond that, I don't see a good reason to keep `init` around.

In another PR I can remove the incremental compilation piece and we should be able to remove init.

Reply:
Ah, I was conflating command-line arguments vs. the arguments supplied with each worker task. Carry on.